replayserver/bookkeeping/bookkeeper.py

```diff
@@ -32,5 +37,27 @@ def build(cls, database, config):
             await self._saver.save_replay(game_id, stream)
             logger.debug(f"Saved replay {game_id}")
             metrics.saved_replays.inc()
+            logger.debug(f"Parsing replay {game_id}")
+            ticks = await self.get_ticks(stream)
+            await self._queries.update_game_stats(game_id, ticks)
         except BookkeepingError as e:
             logger.warning(f"Failed to save replay for game {game_id}: {e}")
+
+    async def get_ticks(self, stream):
+        """Parse the replay and extract the total number of ticks"""
+        return _get_replay_ticks(stream.data.bytes())
+
+
+def _get_replay_ticks(data):
+    reader = ReplayReader(data)
```

Comment:
`ReplayBody` without header won't work.
There https://github.com/FAForever/faf-scfa-replay-parser/blob/master/tests/unittests/replay.py#L82

Reply:
Just use that method and give the data to `parse`.

Reply:
The data here is only the replay body. At this point the server has already parsed the header and split the data into two parts